### ISSUE TITLE
MongoDB Always try to connect on localhost

### ISSUE THREAD

**Issue by wildsonc** [n8n-io/n8n#6878]
**Describe the bug**
Using connection string or raw values always try to connect on localhost, I never managed to use this node, even in very old versions <200.

![image](https://github.com/n8n-io/n8n/assets/72408418/87d5cca5-1e1b-4530-842a-0751aad3d89d)

**To Reproduce**

1. Create credential
2. Click on Save
3. See error

**Environment (please complete the following information):**

- OS: Docker linux
- n8n Version 1.1.1
- Database system PostgreSQL
- Operation mode `queue`

**Comment by Joffcom** [comment]
Hey @wildsonc,

That is odd I have just given it a quick go on my n8n install and it is working as expected.

![image](https://github.com/n8n-io/n8n/assets/4688521/5c8eb1df-9275-4386-a6da-9f5e8e2b375b)

Out of interest what is the network mode for your n8n container? Is it possible that it can't reach your MongoDB install?

**Comment by wildsonc** [comment]
Yes, I have a metabase instance on the same machine, and through it I can make the connection. I'm currently using metabase node to do these mongo queries. I have connections to other types of databases in n8n, so I don't think it's a container network problem.

![image](https://github.com/n8n-io/n8n/assets/72408418/55ec4c9e-667e-4d02-9a99-9bbef83c9680)

I also tested it with a connection to mongo atlas and it worked perfectly, I really don't know what it could be.

I tested with the mongo+srv driver, here the error was already different but it seems to me to be an error on the mongo side?

![image](https://github.com/n8n-io/n8n/assets/72408418/a367498b-6d0f-45cb-9f46-5403b3c1d94b)

**Comment by Joffcom** [comment]
Hey @wildsonc,

That is very odd, I would have thought it would be easy to reproduce if it was a bug. When you say you have metabase on the same machine what do you mean? I would assume if you have n8n in Docker you wouldn't have Metabase in the same container unless you have created a custom image so I wouldn't rule out it being an issue with the Docker networking.

Are you running your MongoDB instance in a container as well?

**Comment by wildsonc** [comment]
@Joffcom It's on a dedicated machine.

In the logs I could see the connection attempt, but I didn't see any log of why it didn't connect:

`2023-08-08T14:45:39.526-0300 I NETWORK  [conn37907] received client metadata from <n8n-ip>:43034 conn37907: { driver: { name: "nodejs", version: "4.16.0" }, platform: "Node.js v16.20.1, LE", os: { name: "linux", architecture: "x64", version: "5.15.0-69-generic", type: "Linux" } }`

**Comment by Joffcom** [comment]
Hey @wildsonc,

That is very strange, I wonder if Mongo is returning something unexpected.

**Comment by Joffcom** [comment]
Hey @wildsonc,

Did you manage to find anything else in the MongoDB logs? I have still not been able to reproduce this.

**Comment by wildsonc** [comment]
Hello @Joffcom ,

I couldn't, maybe it's just something in my installation, even though I managed to connect to other applications (DataGrip and Metabase). I ended up using the metabase as an intermediary.

**Comment by Joffcom** [comment]
For a bit of science would you be able to set up a clean mongodb instance in a container and see if that works? That would help show if it is maybe a config issue, I think atlas working is a good sign that it could be something local.

It is just going to bug me if we can't work it out 🙂

**Comment by wildsonc** [comment]
I created a mongodb in docker and managed to connect. I manually tested connecting using mongodb and mongoose and had the same problem as **ECONNREFUSED**, via the command line (linux) I was also able to connect, just using node I could not make the connection.

I'm going to close this issue because I have no doubts that it's not an n8n problem, but something between my instance and specific nodejs clients.

Thanks @Joffcom.
